### ISSUE TITLE
NullReferenceException in LayoutAnchorable.ToggleAutoHide() after layout deserialisation - Starting in v3.2.0 

### ISSUE THREAD

**Issue by swythan** [xceedsoftware/wpftoolkit#1192]
Version 3.2.0 seems to have introduced a problem in the layout deserialization.

To hit the problem:
1. Load a docking layout that has empty elements for LeftSide, RightSide, etc. and an Anchorable in a layoutAnchorablePane.
1. Click the Auto-Hide button on the Anchorable

I successfully reproduced the bug by running this code in the Loaded event of the AvalonDockView in the LiveExplorer demo code (and applying x:Name="dockManager" to the DockingManager in the XAML):
```
string layoutXml = 
@"<?xml version=""1.0""?>
<LayoutRoot>
  <RootPanel Orientation=""Horizontal"" >
    <LayoutDocumentPane />
    <LayoutAnchorablePane DockWidth=""198"" >
      <LayoutAnchorable Title=""Anchorable"" />
    </LayoutAnchorablePane >
  </RootPanel>
  <TopSide />
  <RightSide />
  <LeftSide />
  <BottomSide />
  <FloatingWindows />
  <Hidden />
</LayoutRoot>";

// Apply the saved docking layout
using (var reader = new StringReader(layoutXml))
{
    var serializer = new XmlLayoutSerializer(this.dockManager);

    serializer.LayoutSerializationCallback += (s, args) => { };
    serializer.Deserialize(reader);
}
```
Clicking Auto-Hide gets me the following exception:
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Xceed.Wpf.AvalonDock.Layout.LayoutAnchorable.ToggleAutoHide() in D:\Source\Repos\WPFToolkit\ExtendedWPFToolkitSolution\Src\Xceed.Wpf.AvalonDock\Layout\LayoutAnchorable.cs:line 538
   at Xceed.Wpf.AvalonDock.DockingManager._ExecuteAutoHideCommand(LayoutAnchorable _anchorable) in D:\Source\Repos\WPFToolkit\ExtendedWPFToolkitSolution\Src\Xceed.Wpf.AvalonDock\DockingManager.cs:line 2556
   at Xceed.Wpf.AvalonDock.Controls.LayoutAnchorableItem.ExecuteAutoHideCommand(Object parameter) in D:\Source\Repos\WPFToolkit\ExtendedWPFToolkitSolution\Src\Xceed.Wpf.AvalonDock\Controls\LayoutAnchorableItem.cs:line 219
   at Xceed.Wpf.AvalonDock.Controls.LayoutAnchorableItem.<InitDefaultCommands>b__8_2(Object p) in D:\Source\Repos\WPFToolkit\ExtendedWPFToolkitSolution\Src\Xceed.Wpf.AvalonDock\Controls\LayoutAnchorableItem.cs:line 68
   at Xceed.Wpf.AvalonDock.Commands.RelayCommand.Execute(Object parameter) in D:\Source\Repos\WPFToolkit\ExtendedWPFToolkitSolution\Src\Xceed.Wpf.AvalonDock\Commands\RelayCommand.cs:line 66
   at MS.Internal.Commands.CommandHelpers.CriticalExecuteCommandSource(ICommandSource commandSource, Boolean userInitiated)
   at System.Windows.Controls.Primitives.ButtonBase.OnClick()
   at System.Windows.Controls.Button.OnClick()
   at System.Windows.Controls.Primitives.ButtonBase.OnMouseLeftButtonUp(MouseButtonEventArgs e)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
<snip>
```

The cause of the NullReferenceException is that LayoutRoot.LeftSide, LayoutRoot.RightSide, etc. are null after deserialization. It looks like the underlying cause is this code (specifically the null return):
https://github.com/xceedsoftware/wpftoolkit/blob/71b8d156d57660f01719e71f121c97d3baacdbbb/ExtendedWPFToolkitSolution/Src/Xceed.Wpf.AvalonDock/Layout/LayoutRoot.cs#L856-L866

I tried setting LayoutRoot.LeftSide, etc. to new LayoutAnchorSide instances after deserialization, but that led to other issues (e.g. auto-hide made the anchorable completely disappear). 

For now I've reverted back to v3.1.0, and that works fine.

**Comment by XceedBoucherS** [comment]
Hi,

Thanks for the well explained description. 
Sorry about that. It is already fixed in the v3.3 and over versions. 
Just to let you know, the Toolkit v3.5 Plus version is available to buy here : https://xceed.com/xceed-toolkit-plus-for-wpf/.

You can wait for v3.3 to come out, buy the v3.5 Plus version or do the following fix :
In file Xceed.Wpf.AvalonDock/Layout/LayoutRoot.cs
Replace the method "ReadXml" with the attached code.
[1192.txt](https://github.com/xceedsoftware/wpftoolkit/files/1344421/1192.txt)

**Comment by nikolayarhangelov** [comment]
When do you plan or releasing v3.3 of the toolkit?

**Comment by XceedBoucherS** [comment]
Hi,
Toolkit Plus v3.6 should be released in early 2018, so v3.3 OpenSource should be released at about the same time.
